### DISCUSSION TITLE
Multiple Firestore databases

### DISCUSSION THREAD

**Discussion by damithg-dev** [invertase/react-native-firebase#7337] [General]
Recently, Firebase introduced Multiple Firestore databases for projects; this will be helpful for someone who is looking for region Firestore.

**Comment by ori9998** [comment]
Is there a example how to use multiple databases? I've created multiple databases but wondering how to database when query.

**Reply by damithg-dev** [reply]
It's not supported yet, but I plan to work on it.

**Reply by ori9998** [reply]
Can I know the expected d-day for supporting multiple databases?

**Reply by mikehardy** [reply]
@ori9998 this is an open source repository. There are no expectations or deadlines as they are meaningless in an open source context - things get done in general when people need things enough that they invest the time to implement them and contribute a PR.

That said, this discussion is based on some abstract text, no directly links to the announcement that I could track down (https://firebase.blog/category/firestore) or links to APIs in the firebase-js-sdk (https://firebase.google.com/docs/reference/js/firestore_)

...so I'm not sure really what the request is even

**Reply by ori9998** [reply]
Oh yes certainly didn't mean that I am expecting some hard 'deadline' like what companies do. I was just checking up whether any of the maintainers have set up some sort of internal due dates/targets. Thanks for clarification.

Besides, as you've mentioned, I've looked up firebase js, functions, storage and such, trying to implement multiple database to react native. It would be great if I could've done it and do a pull request, but I think I met my limit. Hopefully, someone better than me would implement this feature for react native developers. Thank you for the references!

**Reply by mikehardy** [reply]
Finally found the announcement(s) at least, so that's something

https://www.youtube.com/watch?v=bTjLidwxsm0#t=54s
https://cloud.google.com/blog/products/databases/manage-multiple-firestore-databases-in-a-project

These are the relevant firebase-js-sdk APIs to mimic:
https://github.com/firebase/firebase-js-sdk/pull/7356/files#diff-6aeed3e4322a865117ea0d85548a58f76246db16c70deb5545cde87c959464e3

(there are some constraints on, for example, what you may use as a valid database name when creating them, spelled out in the management doc for multidb firestore https://cloud.google.com/firestore/docs/manage-databases)

And the typical process is to add those to the javascript (and type files) in lib in the packages here, mapping them down to the native APIs in the ios and android code while referring to the firebase reference documentation to the native SDK - typically done by adding e2e tests (marked `.only` during dev...) that exercise the JS apis from lib and reloading the e2e app in a code/run/inspect work loop until everything looks right

**Reply by arndtvoges** [reply]
> It's not supported yet, but I plan to work on it. 
> 
Is that still the case? Otherwise we will consider putting forward a PR.

**Reply by anurag-lio** [reply]
@arndtvoges were you guys able to work on this? Is there a PR open?

**Reply by ktanoooo** [reply]
The option to specify a non-default database name has already been merged (https://github.com/invertase/react-native-firebase/pull/7949). 
It worked when I wrote it as follows.

```jsx
import { getApp } from '@react-native-firebase/app';
import { getFirestore } from '@react-native-firebase/firestore';

getFirestore(getApp(), 'yourDatabaseName')
  .collection('yourCollectionName')
  .doc('yourDocumentId')
  .onSnapshot((documentSnapshot) => {
    console.log('data: ', documentSnapshot.data());
  });
```
